### PR TITLE
Fix compilation on K.CompilationUnit.service

### PR DESCRIPTION
**Pull request by timtebeek**
After https://github.com/openrewrite/rewrite/commit/33d8f7d42f9bd6749ab93171fad31289b0e2bc97

### CODE REVIEW

**Review by pstreef — APPROVED** [review]

File: src/main/java/org/openrewrite/kotlin/tree/K.java [hunk 1]
```diff
@@ -341,7 +341,7 @@ public K.CompilationUnit withStatements(List<JRightPadded<Statement>> statements
 
         @Override
         @SuppressWarnings("unchecked")
-        public <S> S service(Class<S> service) {
+        public <S, T extends S> T service(Class<S> service) {
             String serviceName = service.getName();
             try {
                 Class<S> serviceClass;
```
[the remark this answers is not in the document]
Comment (pstreef) [reply src/main/java/org/openrewrite/kotlin/tree/K.java:347]: I think it should. As T extends S and you are returning T you can cast T to S but not S to T

[the remark this answers is not in the document]
Comment (timtebeek) [reply src/main/java/org/openrewrite/kotlin/tree/K.java:347]: @sambsnyd any insight here appreciated; wouldn't want to hide a compile error with a misplaced cast.

**Review by sambsnyd — APPROVED** [review]

File: src/main/java/org/openrewrite/kotlin/tree/K.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sambsnyd) [reply src/main/java/org/openrewrite/kotlin/tree/K.java:347]: I might not have needed to change this signature originally. Whether it returns `S` or `T extends S` the meaning, and opportunities for `ClassCastException` are largely the same.

**Comment by timtebeek** [comment]
Pushed a similar change to rewrite-javascript such that does not fall out of line with our releases https://github.com/openrewrite/rewrite-javascript/commit/8a6d7e8ea17aed67705928a0651a9cacf76e02de

**Pull request merged**